packages-exp/auth-exp/src/core/persistence/browser.ts

```diff
@@ -56,9 +56,28 @@ class BrowserPersistence implements Persistence {
   }
 }
 
-export const browserLocalPersistence: externs.Persistence = new BrowserPersistence(
-  localStorage
-);
-export const browserSessionPersistence: externs.Persistence = new BrowserPersistence(
-  sessionStorage
-);
+class BrowserLocalPersistence {
+  static type: 'LOCAL' = 'LOCAL';
+  static instance: Persistence | null = null;
+  static _getInstance(): Persistence {
```

Comment:
can you make this a generic function and just implement once on PersistenceInstantiator? `_getInstance<T>(): T`

Reply:
Done